packages/blade/src/components/TopNav/TopNav.stories.tsx

```diff
@@ -0,0 +1,292 @@
+  props: Omit<TabNavItemProps, 'as'> & {
+    activeOnLinks?: string[];
+  },
+): React.ReactElement => {
+  const location = useLocation();
+
+  return (
+    <TabNavItem
+      {...props}
+      as={Link}
+      isActive={isItemActive(location, { href: props.href, activeOnLinks: props.activeOnLinks })}
+    />
+  );
+};
+
+const ExploreItem = ({
+  icon: Icon,
+  title,
+  description,
+}: {
+  icon: IconComponent;
+  title: string;
+  description: string;
+}): React.ReactElement => {
+  return (
+    <Box display="flex" gap="spacing.4">
+      <Box
+        borderRadius="medium"
+        padding="spacing.5"
+        backgroundColor="surface.background.gray.subtle"
+      >
+        <Icon color="interactive.icon.neutral.subtle" size="medium" />
+      </Box>
+      <Box>
+        <Text color="surface.text.gray.subtle" size="medium" weight="semibold">
+          {title}
+        </Text>
+        <Text size="small" color="surface.text.gray.muted">
+          {description}
+        </Text>
+      </Box>
+    </Box>
+  );
+};
+
+const TopNavTemplate: StoryFn<typeof TopNav> = () => {
+  const [isSideBarOpen, setIsSideBarOpen] = React.useState(false);
+  const isMobile = useIsMobile();
+  const [selectedProduct, setSelectedProduct] = React.useState<string | null>(null);
+
+  return (
+    <Box>
+      <TopNav>
+        {/* TopNavBrand gets hidden on mobile */}
+        <TopNavBrand>
+          <RazorpayLogo />
+        </TopNavBrand>
+        <TopNavContent>
+          {/* Desktop */}
+          <TabNav display={{ base: 'none', m: 'flex' }}>
+            <TabNavItemLink leading={HomeIcon} accessibilityLabel="Home" href="/home" />
+            <TabNavItemLink href="/payroll">Payroll</TabNavItemLink>
+            <TabNavItemLink href="/payments">Payments</TabNavItemLink>
+            <TabNavItemLink href="/magic-checkout">Magic Checkout</TabNavItemLink>
+            <Menu openInteraction="hover">
+              <TabNavItem href="#">
+                {selectedProduct ? `Explore: ${selectedProduct}` : 'Explore'}
+              </TabNavItem>
+              <MenuOverlay>
+                <MenuHeader
+                  title="Products for you"
+                  trailing={
+                    <Badge emphasis="subtle" color="notice">
+                      Recommended
+                    </Badge>
+                  }
+                />
+                <MenuItem onClick={() => setSelectedProduct('Payroll')}>
+                  <ExploreItem
+                    icon={RazorpayxPayrollIcon}
+                    title="Payroll"
+                    description="Supercharge your process of paying salaries to your employees"
+                  />
+                </MenuItem>
+                <MenuItem onClick={() => setSelectedProduct('Payout')}>
+                  <ExploreItem
+                    icon={BulkPayoutsIcon}
+                    title="Payout"
+                    description="Pay your vendors at ease with RazorpayX"
+                  />
```

Comment:
Nit: Fix it if you have time but can skip this as well. The alignment is strange in example 

<img width="534" alt="image" src="https://github.com/razorpay/blade/assets/30949385/152a0821-0214-4d9e-b978-cb41bf3da5d9">